src/lib.rs

```diff
@@ -120,17 +126,223 @@ fn footprint_latitude(e1: f64, mu: f64) -> f64 {
         + term4 * (8.0 * mu).sin()
 }
 
+const K0: f64 = 0.9996;
+const E: f64 = 0.00669438;
+
+/// Converts a UTM coordinate to a latitude and longitude.
+/// zone_num can be obtain by calling lat_lon_to_zone_number
+/// zone_letter can be obtain by calling lat_to_zone_letter
+///
+/// # Example
+///
+/// ```
+/// use utm::wsg84_utm_to_lat_lon;
+/// const DELTA: f64 = 3e-5;
+/// fn is_close(a: f64, b: f64, epsilon: f64) -> bool {
+///        (a - b).abs() < epsilon
+/// }
+/// // Capetown, South Africa,
+/// let easting = 261878_f64;
+/// let northing = 6243186_f64;
+/// let zone_num = 34_u8;
+/// let zone_letter = 'H';
+/// let (lat, long) = wsg84_utm_to_lat_lon(easting, northing, zone_num, zone_letter);
+/// assert_eq!(is_close(lat, -33.92487, DELTA), true);
+/// assert_eq!(is_close(long, 18.42406, DELTA), true);
+/// ```
+pub fn wsg84_utm_to_lat_lon(
+    easting: f64,
+    northing: f64,
+    zone_num: u8,
+    zone_letter: char,
+) -> (f64, f64) {
+    if easting < 100000. || 1000000. <= easting {
+        panic!("Easting must be between 100000 and 999999");
+    }
+    if northing < 0. || northing > 10000000. {
+        panic!("Northing must be between 0 and 10000000");
+    }
+    if zone_num < 1 || zone_num > 60 {
+        panic!("Zone number must be between 1 and 60");
+    }
+    if zone_letter < 'C' || zone_letter > 'X' {
+        panic!("Zone letter must be between C and X");
+    }
```

Comment:
We should return a `Result<(f64, f64), Error>` to allow users to gracefully recover from exceptional cases. I like to use https://docs.rs/thiserror/latest/thiserror/ for custom error enums.

Reply:
I've updated the code to allow users to recover from the cases and avoid panicking. 

I'm relatively new to rust and I don't really understand the principle of the `no_std` flag. I'm asking myself if the `thiserror` lib will not break the `no_std` flag usage ?

I've tested the library using `cargo test --features no_std` and it breaks with the result 
```shell
error[E0433]: failed to resolve: could not find `error` in `std`
```

If this is a problem, I can revert to avoid using `thiserror`

Reply:
`no_std` is crate-level attribute to disable any platform-specific components of the std crate. It is useful in embedded systems, for more info: https://docs.rust-embedded.org/book/intro/no-std.html.

It looks like `thiserror` doesn't support `no_std` and probably won't: https://github.com/dtolnay/thiserror/pull/64. Since this is just a single error enum for a single case, I recommend:
1. Don't use `thiserror`
2. Make an `enum Error {}` to hold your error variants
3. Use a feature-gate to `impl std::error::Error for Error` only when *not* in `no_std`

If you have issues implementing, flag me and I'll see if I have some moments to take a look myself.